### PR TITLE
Remove light weight server sdk code and add release scripts

### PR DESCRIPTION
**Pull request by wenxiul**
*Issue #, if available:* we shouldn't distribute ddl/lib/so files in the repo. We also don't want to duplicate the light weight server sdk source code which could be downloaded at [Amazon GameLift's Getting Started](https://aws.amazon.com/gamelift/getting-started/).

*Description of changes:* 

- Add release scripts that packages the plugin for release. 
- Delete library files and light weight server sdk source code. 
- Updated README.md.

running release script `export.ps1` generates file structure- 

amazon-gamelift-plugin-unreal -> GameliftPlugin -> build -> amazon-gamelift-plugin-unreal-release-\<plugin version>.zip, which contains three items: 

- README.md
- amazon-gamelift-plugin-unreal-<version>-sdk-<lw-sdk-version>.zip(zip the whole unreal github repo)
- GameLift-Cpp-ServerSDK-<sdk-version>.zip (the c++ server sdk downloaded)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by AWSJackson — COMMENTED** [review]
Haven't had a chance to look at the scripts, so here are my comments on the readmes

File: GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+$TEMP_EXTRACTED_PATH="$TEMP_PATH\GameLiftServer"
+$ITEMS_TO_COPY_PATH="$TEMP_EXTRACTED_PATH\GameLiftServerSDK\*"
+$SERVER_SDK_S3_LINK_PREFIX="https://gamelift-server-sdk-release.s3.us-west-2.amazonaws.com/unreal"
+$SERVER_SDK_S3_DOWNLOAD_LINK="$SERVER_SDK_S3_LINK_PREFIX/$SERVER_SDK_FILENAME"
+$PACKAGED_ZIP_PATH="$BUILD_DIR\amazon-gamelift-plugin-unreal-$PluginVersion-sdk-$ServerSdkVersion.zip"
+$LISCENSE_FILE_PATH="$TEMP_EXTRACTED_PATH\GameLiftServerSDK\LICENSE_AMAZON_GAMELIFT_SDK.TXT"
+
+if (Test-Path -Path $PACKAGED_ZIP_PATH)
+{
+	Write-Host "Amazon GameLift Server SDK zip has already been prepared, skipping download and extraction." -ForegroundColor DarkYellow
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1:29]: Please update these comments to specify that this is the Lightweight Unreal Server SDK so there's no confusion with the C++ server sdk

File: GameLiftPlugin/Scripts~/windows/.export-cpp-sdk-zip.ps1 [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+{
+	Write-Host "Downloading Amazon GameLift C++ SDK $ServerSdkVersion..."
+	# Download link should be public and require no credentials
+	iwr $CPP_SERVER_SDK_S3_DOWNLOAD_LINK -OutFile $CPP_SERVER_SDK_TEMP_ZIP_PATH
+}
+else
+{
+	Write-Host "Amazon GameLift C++ Server SDK is already downloaded and stored at $CPP_SERVER_SDK_TEMP_ZIP_PATH, skipping download."
+}
+Write-Host "Removing liscense file..."
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.export-cpp-sdk-zip.ps1:35]: minor: typo `license` not `liscense` (a couple places in this PR)

File: GameLiftPlugin/Scripts~/windows/.export-release.ps1 [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	[string] $ServerSdkVersion,
+	[Parameter(Mandatory)]
+	[ValidatePattern("\d+\.\d+\.\d+")]
+	[string] $PluginVersion
+)
+
+$ROOT_DIR=Resolve-Path "$PSScriptRoot\..\..\.."
+$BUILD_DIR="$ROOT_DIR\GameLiftPlugin\build"
+$STAGING_PATH="$BUILD_DIR\staging-release"
+$README_PATH=Resolve-Path "$PSScriptRoot\..\PACKAGE_FOR_RELEASE.md"
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.export-release.ps1:14]: minor: IMO it's better to base this off the ROOT_DIR so that if this script moves only the root directory needs to be updated

**Review by AWSJackson — CHANGES_REQUESTED** [review]
Can you attach to this PR an example release zip when running this script?

File: GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    mkdir out
+    cd out
+    cmake -DBUILD_FOR_UNREAL=1 ..
+    make
+    ```
+1. Once complete, the following file should have been built
+    ```
+    prefix/lib/aws-cpp-sdk-gamelift-server.so
+    ```
+1. Copy the file over to this location in the Unreal plugin folder. Note if you choose to clone the plugin from repo, you need to download the C++ Server SDK Plugin for Unreal from [Amazon GameLift's Getting Started](https://aws.amazon.com/gamelift/getting-started/), then copy the unzipped GameLiftServerSDK folder under `GameLiftPlugin/Source/` and rename the folder `GameLiftServerSDK` to `GameLiftServer`.
```
[the remark this answers is not in the document]
Comment (wenxiul) [reply GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md:28]: modified as suggested in next revision

File: GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+------
+4.22  
+4.23  
+4.24  
+4.25  
+4.26  
+4.27   
+5.1.0  
+5.1.1
```
[the remark this answers is not in the document]
Comment (wenxiul) [reply GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md:77]: moved to top of readme. Our plugin readme says compatible with UE 5.3 as well. So I also added 5.3

File: GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1 [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+{
+	Write-Host "Amazon GameLift Server SDK zip has already been unpacked at $DESTINATION_PATH, skipping unpack" -ForegroundColor DarkYellow
+}
+else
+{
+	Write-Host "Extracting into project directory..."
+	Expand-Archive -Force -LiteralPath $TEMP_ZIP_PATH -DestinationPath $TEMP_EXTRACTED_PATH -ErrorAction Stop
+	Remove-Item -Force -LiteralPath $LISCENSE_FILE_PATH -ErrorAction Stop
+	New-Item -Force -Type Directory -Path $DESTINATION_PATH -ErrorAction Stop
+	Copy-Item -Path $ITEMS_TO_COPY_PATH -Destination $DESTINATION_PATH -Recurse
```
[the remark this answers is not in the document]
Comment (wenxiul) [reply GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1:55]: added DESTINATION_PATH\* to .gitignore, except for the readme files under that dir since we're adding a readme directing customers to put lightweight source files there.

File: GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1 [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+New-Item -Force -Type Directory -Path $BUILD_DIR -ErrorAction Stop
+if (Test-Path -Path $TEMP_PLUGIN_PATH)
+{
+	Write-Host "Amazon GameLift Server SDK zip has already been copied to $TEMP_PLUGIN_PATH, skipping copy" -ForegroundColor DarkYellow
+}
+else
+{
+	New-Item -Force -Type Directory -Path $TEMP_PLUGIN_PATH -ErrorAction Stop
+	Copy-Item -Path "$ROOT_DIR\*" -Destination $TEMP_PLUGIN_PATH -Recurse -Include *.md,GameLiftPlugin -ErrorAction Stop
+}
```
[the remark this answers is not in the document]
Comment (wenxiul) [reply GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1:67]: when compressing in place, i got this error- ZipArchiveHelper : The process cannot access the file 'C:\Users\Administrator\Desktop\forpr\amazon-gamelift-plugin-unreal\.vs\amazon-gamelift-plugin-unreal\FileContentIndex\16852b0a-76f7-4b52-973b-2271971c2471.vsidx' because it is being used by
another process. seems like an error from compressing the directory where this script itself is in.

File: GameLiftPlugin/Scripts~/windows/export.ps1
[the remark this answers is not in the document]
Comment (wenxiul) [reply GameLiftPlugin/Scripts~/windows/export.ps1:1]: added .clean-download-files.ps1 and .clean-exports.ps1

File: GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
+# GameLiftServerSDK Unreal Plugin
+## Documention
+You can find the official Amazon GameLift documentation [here](https://aws.amazon.com/documentation/gamelift/).
+
+
+## Using the plugin
+### Building the C++ SDK
+Before we can use the plugin inside an Unreal game, we need to compile the Amazon GameLift server C++ SDK.
```
Comment (amayreh-gamelift) [inline GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md:9]: - In the next line the instructions say "To **build** the ...", in this line it says "need to **compile** the ..."
My suggestion is to change this to "need to **build** the ...".
- Instead of "We", I would use "You" like this >> "Before you can use the plugin inside an Unreal game, you need to build the Amazon GameLift server C++ SDK."

File: GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+2. Navigate to the `GameLift-Cpp-ServerSDK-<version>` folder that was included with the Amazon GameLift SDK Release download.
+1. Follow the below instructions for your platform.  
+
+#### Linux
+
+1. Run the following commands
+    ```sh
+    mkdir out
+    cd out
+    cmake -DBUILD_FOR_UNREAL=1 ..
```
Comment (amayreh-gamelift) [inline GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md:22]: Why are we adding part of the build command here ?

File: GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+    GameLiftPlugin/Source/GameLiftServer/ThirdParty/GameLiftServerSDK/Linux/x86_64-unknown-linux-gnu/aws-cpp-sdk-gamelift-server.so 
+    ```
+
+#### Windows
+
+1. Run the following commands
+    ```sh
+    mkdir out
+    cd out
+    cmake -G "Visual Studio 17 2022" -DBUILD_FOR_UNREAL=1 ..
```
Comment (amayreh-gamelift) [inline GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md:44]: Same here, any reason why we are not writing the full command ?

File: GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md (hunk 8 above)
Comment (wenxiul) [reply GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md]: changed as suggested

File: GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md (hunk 10 above)
Comment (wenxiul) [reply GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md:44]: literally copying this command works. I think the .. is referring to current dir- https://stackoverflow.com/questions/12236642/what-does-cmake-do

**Review by amayreh-gamelift — COMMENTED** [review]
Lets follow readme/instructions step by step, and verify that it works, this way we will be sure that it will work for customers

**Review by amayreh-gamelift — APPROVED** [review]

File: GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (AWSJackson) [reply GameLiftPlugin/Scripts~/PACKAGE_FOR_RELEASE.md]: Please restructure the readme to be similar to the package's main readme (use unity as another example: https://github.com/aws/amazon-gamelift-plugin-unity/blob/develop/Scripts~/PACKAGE_FOR_RELEASE.md)

Example:
```
# Amazon GameLift Plugin for Unreal Engine

![GitHub](https://img.shields.io/github/license/aws/amazon-gamelift-plugin-unreal)
![GitHub release (latest by date)](https://img.shields.io/github/v/release/aws/amazon-gamelift-plugin-unreal)
![GitHub all releases](https://img.shields.io/github/downloads/aws/amazon-gamelift-plugin-unreal/total)
![GitHub release (latest by date)](https://img.shields.io/github/downloads/aws/amazon-gamelift-plugin-unreal/latest/total)

Compatible with Unreal Engine 5 (versions 5.0, 5.1, 5.2 and 5.3).

## Install the plugin

Complete the following steps to install and enable the plugin for your multiplayer game project. For more details, see the [AWS GameLift documentation](https://docs.aws.amazon.com/gamelift/latest/developerguide/unreal-plugin.html).

1. Build the Amazon GameLift C++ Server SDK. See section below for details.

1. Install and enable the plugin.
    1. In your game project root folder, open the `Plugins` folder and copy the Amazon GameLift plugin files.
    1. In the `.uproject` file, add the following to the `Plugins` section:
        ```
        {
            "Name": "GameLiftPlugin",
            "Enabled": true
        },
        {
            "Name": "WebBrowserWidget",
            "Enabled": true
        }
        ```

1. Set your project to use the source-built UE editor. Do this step if your game project was created with a non-source-built version of UE. In your game project folder, select the `.uproject` file and choose the option **Switch Unreal Engine Version**.

1. Rebuild the project solution. After completing the previous steps to update your project files, rebuild the solution.

## Build the Amazon GameLift C++ Server SDK

Before you can use the plugin inside an Unreal game, you need to build the Amazon GameLift C++ Server SDK.  

To build the Amazon GameLift C++ Server SDK:
1. Open a terminal/command prompt.
2. Navigate to the `GameLift-Cpp-ServerSDK-<version>` folder that was included with the Amazon GameLift SDK Release download.
1. Follow the below instructions for your platform.  

### Linux

1. Run the following commands
    ```sh
    mkdir out
    cd out
    cmake -DBUILD_FOR_UNREAL=1 ..
    make
    ```
1. Once complete, the following file should have been built
    ```
    prefix/lib/aws-cpp-sdk-gamelift-server.so
    ```
1. Copy the file over to the following location in the Unreal plugin folder: 
    ```
    GameLiftPlugin/Source/GameLiftServer/ThirdParty/GameLiftServerSDK/Linux/x86_64-unknown-linux-gnu/
    ```  
    Once complete you should have a filepath similar to this example
    ```
    GameLiftPlugin/Source/GameLiftServer/ThirdParty/GameLiftServerSDK/Linux/x86_64-unknown-linux-gnu/aws-cpp-sdk-gamelift-server.so 
    ```

### Windows

1. Run the following commands
    ```sh
    mkdir out
    cd out
    cmake -G "Visual Studio 17 2022" -DBUILD_FOR_UNREAL=1 ..
    msbuild ALL_BUILD.vcxproj /p:Configuration=Release
    ```
1. The above step produces the following binary files required by the plugin.  
    ```
    prefix\bin\aws-cpp-sdk-gamelift-server.dll  
    prefix\lib\aws-cpp-sdk-gamelift-server.lib
    ```
1. Copy the files over to this location in the Unreal plugin folder:
    ```
    GameLiftPlugin\Source\GameLiftServer\ThirdParty\GameLiftServerSDK\Win64\
    ```  
    Once complete you should have two filepaths similar to this example  
    ```
    GameLiftPlugin\Source\GameLiftServer\ThirdParty\GameLiftServerSDK\Win64\aws-cpp-sdk-gamelift-server.dll  
    GameLiftPlugin\Source\GameLiftServer\ThirdParty\GameLiftServerSDK\Win64\aws-cpp-sdk-gamelift-server.lib 
    ```

For more detailed instructions on how to build the C++ Server SDK, please refer to the README.md file located in the C++ Server SDK directory.
```

File: GameLiftPlugin/Scripts~/windows/release.ps1 [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+	[ValidatePattern("\d+\.\d+\.\d+")]
+	[string] $ServerSdkVersion,
+	[Parameter(Mandatory)]
+	[ValidatePattern("\d+\.\d+\.\d+")]
+	[string] $LightWeightUnrealServerSdkVersion
+)
+
+$GAMELIFT_PLUGIN_DIR=Resolve-Path "$PSScriptRoot\..\.."
+Write-Host "Getting version number from 'GameLiftPlugin.uplugin'"
+$PLUGIN_VERSION=Select-String -LiteralPath $GAMELIFT_PLUGIN_DIR\GameLiftPlugin.uplugin -Pattern '"VersionName": ".*",' | % { $_.Matches.Value } | % { $_.substring(16, $_.length-18) }
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/release.ps1:12]: Please add a `Write-Host` indicating what we found for the plugin version

```
Write-Host "Identified plugin version as $PLUGIN_VERSION" -ForegroundColor DarkYellow
```

File: GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+$DESTINATION_PATH="$SOURCE_DIR\GameliftServer"
+$RELEASE_NOTES_PATH="$DESTINATION_PATH\Release-notes"
+$TEMP_PATH="C:\Temp\AmazonGameLiftPluginUnreal"
+$TEMP_ZIP_PATH="$TEMP_PATH\$SERVER_SDK_FILENAME"
+$TEMP_PLUGIN_PATH="$TEMP_PATH\amazon-gamelift-plugin-unreal"
+$TEMP_EXTRACTED_PATH="$TEMP_PATH\GameLiftServer"
+$ITEMS_TO_COPY_PATH="$TEMP_EXTRACTED_PATH\GameLiftServerSDK\*"
+$SERVER_SDK_S3_LINK_PREFIX="https://gamelift-server-sdk-release.s3.us-west-2.amazonaws.com/unreal"
+$SERVER_SDK_S3_DOWNLOAD_LINK="$SERVER_SDK_S3_LINK_PREFIX/$SERVER_SDK_FILENAME"
+$PACKAGED_ZIP_PATH="$BUILD_DIR\amazon-gamelift-plugin-unreal-$PluginVersion-sdk-$LightWeightUnrealServerSdkVersion.zip"
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1:24]: This item doesn't appear relevant to this script

```suggestion
```

File: GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+$ROOT_DIR=Resolve-Path "$PSScriptRoot\..\..\.."
+$SOURCE_DIR="$ROOT_DIR\GameLiftPlugin\Source"
+$BUILD_DIR="$ROOT_DIR\GameLiftPlugin\build"
+$SERVER_SDK_DIR="GameLift-Cpp-ServerSDK-UnrealPlugin-$LightWeightUnrealServerSdkVersion";
+$SERVER_SDK_FILENAME="$SERVER_SDK_DIR.zip";
+$DESTINATION_PATH="$SOURCE_DIR\GameliftServer"
+$RELEASE_NOTES_PATH="$DESTINATION_PATH\Release-notes"
+$TEMP_PATH="C:\Temp\AmazonGameLiftPluginUnreal"
+$TEMP_ZIP_PATH="$TEMP_PATH\$SERVER_SDK_FILENAME"
+$TEMP_PLUGIN_PATH="$TEMP_PATH\amazon-gamelift-plugin-unreal"
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1:19]: Not used

```suggestion
```

File: GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+$SERVER_SDK_S3_LINK_PREFIX="https://gamelift-server-sdk-release.s3.us-west-2.amazonaws.com/unreal"
+$SERVER_SDK_S3_DOWNLOAD_LINK="$SERVER_SDK_S3_LINK_PREFIX/$SERVER_SDK_FILENAME"
+$PACKAGED_ZIP_PATH="$BUILD_DIR\amazon-gamelift-plugin-unreal-$PluginVersion-sdk-$LightWeightUnrealServerSdkVersion.zip"
+$LICENSE_FILE_PATH="$TEMP_EXTRACTED_PATH\GameLiftServerSDK\LICENSE_AMAZON_GAMELIFT_SDK.TXT"
+
+if (Test-Path -Path $PACKAGED_ZIP_PATH)
+{
+	Write-Host "Amazon GameLift Unreal Plugin zip has already been prepared, skipping packaging." -ForegroundColor DarkYellow
+	exit 0
+}
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1:31]: This item doesn't appear relevant to this script

```suggestion
```

File: GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+{
+	Write-Host "Lightweight Unreal Server SDK for Unreal zip has already been unpacked at $DESTINATION_PATH, skipping unpack" -ForegroundColor DarkYellow
+}
+else
+{
+	Write-Host "Extracting into project directory..."
+	Expand-Archive -Force -LiteralPath $TEMP_ZIP_PATH -DestinationPath $TEMP_EXTRACTED_PATH -ErrorAction Stop
+	Remove-Item -Force -LiteralPath $LICENSE_FILE_PATH -ErrorAction Stop
+	New-Item -Force -Type Directory -Path $DESTINATION_PATH -ErrorAction Stop
+	Copy-Item -Path $ITEMS_TO_COPY_PATH -Destination $DESTINATION_PATH -Recurse
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1:55]: I noticed that the lightweight sdk's readme isn't being imported because the `Copy-Item` is likely not overriding the existing readme. Additionally, any extra files that we don't want there would stick around. I'd suggest either removing all items from the destination directory before the copy, or use a force copy (won't get rid of extra files)

File: GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+	New-Item -Force -Type Directory -Path $TEMP_PATH -ErrorAction Stop
+	Write-Host "Downloading C++ Server SDK Plugin for Unreal $LightWeightUnrealServerSdkVersion..."
+	# Download link should be public and require no credentials
+	iwr $SERVER_SDK_S3_DOWNLOAD_LINK -OutFile $TEMP_ZIP_PATH
+}
+else
+{
+	Write-Host "Lightweight Unreal Server SDK for Unreal zip is already downloaded and stored at $TEMP_ZIP_PATH, skipping download."
+}
+if (Test-Path -Path $RELEASE_NOTES_PATH)
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1:45]: I'm not a huge fan of this check as it relies on the structure of a zip file not controlled by this package. Instead, consider separating unzipping and removing the license from copying the files over (see below comment about force copy and/or remove then copy)

```
if (Test-Path -Path $TEMP_EXTRACTED_PATH)
{
	Write-Host "Lightweight Server SDK for Unreal zip has already been unpacked at $DESTINATION_PATH, skipping unpack" -ForegroundColor DarkYellow
}
else
{
	Expand-Archive -Force -LiteralPath $TEMP_ZIP_PATH -DestinationPath $TEMP_EXTRACTED_PATH -ErrorAction Stop
	Remove-Item -Force -LiteralPath $LICENSE_FILE_PATH -ErrorAction Stop
}

Write-Host "Importing Lightweight Server SDK for Unreal source into $DESTINATION_PATH"
New-Item -Force -Type Directory -Path $DESTINATION_PATH -ErrorAction Stop
Copy-Item -Path $ITEMS_TO_COPY_PATH -Destination $DESTINATION_PATH -Recurse -Force

exit 0
```

File: GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+if (-Not (Test-Path -Path $TEMP_ZIP_PATH) )
+{
+	New-Item -Force -Type Directory -Path $TEMP_PATH -ErrorAction Stop
+	Write-Host "Downloading C++ Server SDK Plugin for Unreal $LightWeightUnrealServerSdkVersion..."
+	# Download link should be public and require no credentials
+	iwr $SERVER_SDK_S3_DOWNLOAD_LINK -OutFile $TEMP_ZIP_PATH
+}
+else
+{
+	Write-Host "Lightweight Unreal Server SDK for Unreal zip is already downloaded and stored at $TEMP_ZIP_PATH, skipping download."
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1:43]: minor: says Unreal twice (multiple occurrences in this file)
```suggestion
	Write-Host "Lightweight Server SDK for Unreal zip is already downloaded and stored at $TEMP_ZIP_PATH, skipping download."
```

File: GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 18]
```diff
@@ -0,0 +1,57 @@
+param(
+	[Parameter(Mandatory)]
+	[ValidatePattern("\d+\.\d+\.\d+")]
+	[string] $LightWeightUnrealServerSdkVersion,
+	[Parameter(Mandatory)]
+	[ValidatePattern("\d+\.\d+\.\d+")]
+	[string] $PluginVersion
```
Comment (AWSJackson) [inline GameLiftPlugin/Scripts~/windows/.import-lightweight-sdk.ps1:7]: See below comments

```suggestion
```

File: GameLiftPlugin/Scripts~/windows/export.ps1
[the remark this answers is not in the document]
Comment (AWSJackson) [reply GameLiftPlugin/Scripts~/windows/export.ps1:1]: We'd also want something to cleanup the imported source code from the lightweight server sdk

File: GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1 (hunk 7 above)
[the remark this answers is not in the document]
Comment (AWSJackson) [reply GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1]: Gotcha. Can you include a comment that we're `# Using a temporary directory since zipping in place conflicts with other processes accessing the files including itself`.

File: README.md [hunk 19]
```diff
@@ -23,14 +23,74 @@ Each scenario uses an AWS CloudFormation template to  deploy your game, creating
...
+* C++ Server SDK
 * Microsoft Visual Studio 2019 or newer.
 * A source-built version of the Unreal Engine editor. Required to develop server build components for a multiplayer game. See the Unreal Engine documentation: 
     * [Accessing Unreal Engine source code on GitHub](https://www.unrealengine.com/ue-on-github). Requires  GitHub and Epic Games accounts.
     * [Building Unreal Engine from Source](https://docs.unrealengine.com/5.3/en-US/building-unreal-engine-from-source/) 
 * (Optional) A C++ multiplayer game project with game code. Projects that use Blueprints only are not compatible with this plugin, at this time.
 * An AWS account with access permissions to use AWS GameLift. See [Set up programmatic access](https://docs.aws.amazon.com/gamelift/latest/developerguide/setting-up-aws-login.html) with long-term credentials.
 
+## Using the plugin
+### Building the C++ SDK
```
Comment (AWSJackson) [inline README.md:38]: See comment on packaging for release readme

File: Scripts~/windows/.export-plugin-zip.ps1 [hunk 20]
```diff
@@ -0,0 +1,28 @@
...
+	[string] $PluginVersion
+)
+
+$ROOT_DIR=Resolve-Path "$PSScriptRoot\..\.."
+$BUILD_DIR="$ROOT_DIR\GameLiftPlugin\build"
+$TEMP_PATH="C:\Temp\AmazonGameLiftPluginUnreal"
+$TEMP_PLUGIN_PATH="$TEMP_PATH\amazon-gamelift-plugin-unreal"
+$PACKAGED_ZIP_PATH="$BUILD_DIR\amazon-gamelift-plugin-unreal-$PluginVersion-sdk-$LightWeightUnrealServerSdkVersion.zip"
+
+New-Item -Force -Type Directory -Path $BUILD_DIR -ErrorAction Stop
```
Comment (AWSJackson) [inline Scripts~/windows/.export-plugin-zip.ps1:16]: minor: move this after the Copy-Item step as this is leaving an empty "build" folder in the resulting release

File: Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+	Write-Host "Lightweight Server SDK for Unreal zip has already been unpacked at $DESTINATION_PATH, skipping unpack" -ForegroundColor DarkYellow
+}
+else
+{
+	Expand-Archive -Force -LiteralPath $TEMP_ZIP_PATH -DestinationPath $TEMP_EXTRACTED_PATH -ErrorAction Stop
+	Remove-Item -Force -LiteralPath $LICENSE_FILE_PATH -ErrorAction Stop
+}
+Write-Host "Importing Lightweight Server SDK for Unreal source into $DESTINATION_PATH"
+New-Item -Force -Type Directory -Path $DESTINATION_PATH -ErrorAction Stop
+Remove-Item "$DESTINATION_PATH\*" -Recurse -Force
```
Comment (AWSJackson) [inline Scripts~/windows/.import-lightweight-sdk.ps1:44]: Can we add a comment explaining briefly why we're doing this?

File: GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1 (hunk 6 above)
[the remark this answers is not in the document]
Comment (AWSJackson) [reply GameLiftPlugin/Scripts~/windows/.export-plugin-zip.ps1]: I'm wondering, we should probably add an explicit clean step that removes all these files. Ideally if I run the clean script by itself, my repo will look like it does when you clone the repo initially (without dropping your intentional development changes)

**Comment by AWSJackson** [comment]
Based on our meeting yesterday, we'll likely re-review the READMEs; however, for now I think they're good to merge

File: GameLiftPlugin/Source/GameliftServer/REPLACEME.md
Comment (AWSJackson) [inline GameLiftPlugin/Source/GameliftServer/REPLACEME.md:1]: minor: I only just noticed, the directory name is `GameliftServer` not `GameLiftServer`

**Review by AWSJackson — APPROVED** [review]

**Review by amayreh-gamelift — APPROVED** [review]

File: Scripts~/windows/.import-lightweight-sdk.ps1 [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+	[Parameter(Mandatory)]
+	[ValidatePattern("\d+\.\d+\.\d+")]
+	[string] $LightWeightUnrealServerSdkVersion
+)
+
+$ROOT_DIR=Resolve-Path "$PSScriptRoot\..\.."
+$SOURCE_DIR="$ROOT_DIR\GameLiftPlugin\Source"
+$SERVER_SDK_DIR="GameLift-Cpp-ServerSDK-UnrealPlugin-$LightWeightUnrealServerSdkVersion";
+$SERVER_SDK_FILENAME="$SERVER_SDK_DIR.zip";
+$DESTINATION_PATH="$SOURCE_DIR\GameliftServer"
```
Comment (AWSJackson) [inline Scripts~/windows/.import-lightweight-sdk.ps1:11]: This is still `GameliftServer`, there may be other places were that's used vs `GameLiftServer`
